### PR TITLE
Add run_as and run_as_user_name field to job definition

### PR DESCRIPTION
**Pull request by dvkuznietsov**
## Changes

Added read-only runAsUserName field to job definition and write-only runAs field to job settings definition. Fields are added in openAPI spec and are marked as private preview.
## Tests
Tested e2e by running `terraform apply` on a dev workspace

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/resources/job.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ The resource supports the following arguments:
 * `email_notifications` - (Optional) (List) An optional set of email addresses notified when runs of this job begins, completes and fails. The default behavior is to not send any emails. This field is a block and is documented below.
 * `webhook_notifications` - (Optional) (List) An optional set of system destinations (for example, webhook destinations or Slack) to be notified when runs of this job begins, completes and fails. The default behavior is to not send any notifications. This field is a block and is documented below.
 * `schedule` - (Optional) (List) An optional periodic schedule for this job. The default behavior is that the job runs when triggered by clicking Run Now in the Jobs UI or sending an API request to runNow. This field is a block and is documented below.
+* `run_as` - (Optional) Optional identifier of the user or service principal that the job runs as. If not specified, the job runs as the user who created the job. This field is a block and is documented below.
```
Comment (gaborratky-db) [inline docs/resources/job.md:91]:
```suggestion
* `run_as` - (Optional) An optional identifier of the user or the service principal that the job runs as. If not specified, the job runs as the user who created the job. This field is a block and is documented below.
```

File: docs/resources/job.md [hunk 2]
```diff
@@ -104,6 +105,24 @@ resource "databricks_job" "this" {
 }
 ```
 
+### run_as Configuration Block
+
+* `user_name` - (Optional) The email of an active workspace user. Non-admin users can only set this field to their own email.
+* `service_principal_name` - (Optional) Application ID of an active service principal. Setting this field requires the `servicePrincipal/user` role.
```
Comment (gaborratky-db) [inline docs/resources/job.md:111]:
```suggestion
* `service_principal_name` - (Optional) The application ID of an active service principal. Setting this field requires the `servicePrincipal/user` role.
```

**Review by gaborratky-db — COMMENTED** [review]
LGTM with nits

File: docs/resources/job.md [hunk 3]
```diff
@@ -88,6 +88,7 @@ The resource supports the following arguments:
 * `email_notifications` - (Optional) (List) An optional set of email addresses notified when runs of this job begins, completes and fails. The default behavior is to not send any emails. This field is a block and is documented below.
 * `webhook_notifications` - (Optional) (List) An optional set of system destinations (for example, webhook destinations or Slack) to be notified when runs of this job begins, completes and fails. The default behavior is to not send any notifications. This field is a block and is documented below.
 * `schedule` - (Optional) (List) An optional periodic schedule for this job. The default behavior is that the job runs when triggered by clicking Run Now in the Jobs UI or sending an API request to runNow. This field is a block and is documented below.
+* `run_as` - (Optional) An optional identifier of the user or the service principal that the job runs as. If not specified, the job runs as the user who created the job. This field is a block and is documented below.
```
Comment (alexott) [inline docs/resources/job.md:91]: Usually this list includes only direct attributes, and blocks are documented as separate sections.

File: jobs/resource_job.go [hunk 4]
```diff
@@ -200,6 +200,11 @@ type ContinuousConf struct {
 type Queue struct {
 }
 
+type JobRunAs struct {
+	UserName             string `json:"user_name,omitempty"`
+	ServicePrincipalName string `json:"service_principal_name,omitempty"`
```
Comment (alexott) [inline jobs/resource_job.go:205]: we may need to add a customization to set `ConflictsWith` between these two fields. Also, we may need to add checks that both strings aren't empty...

**Review by alexott — APPROVED** [review]
in general looks good, just few comments

**Comment by alexott** [comment]
Please rebase on the latest `master`...

**Review by nfx — APPROVED** [review]

**Pull request closed without merging**